hack/update-bundles.sh

```diff
@@ -55,36 +91,53 @@ function cleanup() {
 }
 trap cleanup EXIT
 
-
 for b in $BUNDLES; do
   # Find the git sha where the source files were last updated
   mapfile -t src_dirs < <(bundle_src_dirs "$b")
   last_update_sha=$(git log -n 1 --pretty=format:%h -- "${src_dirs[@]}")
 
-  # Check if the bundle for that git sha exists already
   repo=$(repo_name "$b")
   tag=git-$last_update_sha
-  push_repo="${REPO_PREFIX}$repo"
-
-  skopeo_args=()
-  skopeo_cp_args=()
-  if [[ $push_repo == *'localhost:'* ]]; then
-    skopeo_args+=(--tls-verify=false)
-    skopeo_cp_args+=(--dest-tls-verify=false --src-tls-verify=false)
-  fi
 
-  tag_found="$(
-    {
-      skopeo list-tags "${skopeo_args[@]}" "docker://${push_repo}" |
-      jq --arg tag "${tag}" -r 'any(.Tags[]; . == $tag)';
-    } || echo false
-  )"
-  if [[ "$tag_found" == 'true' ]]; then
-    # No push needed
-    echo "Policy bundle $push_repo:$tag exists already, no push needed"
+  # Check all repositories to see if any need the bundle
+  repos_needing_push=()
+  all_repos_have_bundle=true
+
+  for repo_prefix in $REPO_PREFIXES; do
+    push_repo="${repo_prefix}$repo"
+    
+    # Login with correct credentials for tag checking
+    if [[ "$push_repo" == *"enterprise-contract"* ]]; then
```

Comment:
Should the tag checking and pushing be added here also? Seems like it could be done all at once to avoid the duplicate login on push.

Reply:
So the tag checking determines if a given bundle needs to be built, that way it doesn't need to build all bundles every time.